### ISSUE TITLE
Inline SVG support

### ISSUE THREAD

**Issue by Calamitizer** [sghall/react-compound-slider#65]
Hey there! I'm trying to figure out how to use this library with an inline SVG, with each library component rendering separately onto a single root `<svg />`. I think this should be possible, due to the nominally unopinionated nature of compound components, but there seems to be a problem. The issue is two-fold:

1. `<Slider />` wraps its children in (i.e. it renders as) a `<div />`. This means that it can't be within the root `<svg />` element.
2. `<Slider />` expects its *immediate* children to be `<Handles />`, `<Ticks />`, etc. This means that that the root `<svg />` element can't be within it either! (Unless each component gets a separate `<svg />`, which isn't especially desirable.)

It doesn't seem like you can make either way work. Is there anything I'm missing that pokes a hole in that reasoning?

As it stands now, I think that `<div />` might be the single opinion the library does have about rendering. Have you considered adding an optional prop to `<Slider />` that determines the root element? This is similar to `@material-ui/core`'s [`<Typography />`](https://material-ui.com/api/typography/) and its `component` prop. `React.Fragment` might be a nice option as well, since it is truly abstract. Then you could do something like this:

```tsx
<svg
  xmlns={SVG_XMLNS}
  {...{ width, height }}
  viewBox={`0, 0, ${width}, ${height}`}
>
  <Slider
    component={React.Fragment}
    {...otherProps}
  >
    <Ticks>
      {({ ticks }) => (
        <>
          {ticks.map(({ id, percent }) => (
            <g
              className="slider-tick"
              transform={`translate(${percent * width}, ${height / 2})`}
              key={id}
            >
              <path d={/* ... */ } />
            </g>
          ))}
        </>
      )}
    </Ticks>
  </Slider>
</svg>
```

**Comment by sghall** [comment]
Hey. Thanks for opening the issue.  Yeah, you should be able to set the component.  It has to  be string component not `Fragment` because I add a ref and the measurements come off that element.  So for SVG just use a 'g'  element.  Internally the library uses `getBoundingClientRect()`.  Should be pretty quick.

Your pseudo code should basically work, just remember the percent is 0-100 so divide by 100.  Also, you'll need to do a little translating because the handles would hang off the edge in the above code.  I'll try to make a quick sandbox.

**Comment by sghall** [comment]
I think the key to doing an SVG version is to have a prop to render the children as siblings to the slider element.  

I created a new beta version `1.2.0-beta.0` .  You can install it locally.   It has the `component` prop like you suggested and a `siblingChildren` prop (not crazy about this name, maybe "flatten" or something).  You need that because there are no SVG container elements that you can define the width for.  A `g`, for example, won't work because it can be transformed but won't take a width.  A nested SVG will cut off the edges. 

Here's a sandbox using the beta version...
https://codesandbox.io/s/qk0vovqw6

I create a `rect` component with a red outline so you can see it.  That's where the measurements are done from to get the percentages, etc.  But when you have `siblingChildren` set to true all the handles, tracks and ticks re rendered as siblings to the rect element.  Seems to work well.

What do you think?  Comments, suggestions...

Code is on this svg branch
https://github.com/sghall/react-compound-slider/tree/svg

**Comment by Calamitizer** [comment]
Hey, thanks for the impressively quick action on this! Yesterday I upgraded to `1.2.0-beta.0` and played around with it a good bit; so far, it seems like this is enough freedom to do everything with SVG that I was trying to do. Good call with `siblingChildren` — that's a gotcha I didn't foresee. I'm using this in a responsive view and it seems to reflow perfectly.

I'll let you know if any issues or suggestions come up for me as I use it a bit more. Again, thanks for the lightning-fast action; I'm using this in an enterprise context and I was really not looking forward to shimming in a `yarn link`'d replacement.

**Comment by sghall** [comment]
That's great.   Ok, well if you run into any problems let me know.  I'll release it on Monday as 1.2.0 and update this ticket unless something comes up.  I think the only thing I want to change is the `siblingChildren` prop renamed to `flatten`.  I'll link to this ticket in the props documentation so people can understand the context.

**Comment by Calamitizer** [comment]
Sounds good! I made a teeny PR with type definitions too since I'm using TypeScript.

**Comment by sghall** [comment]
Ok.  This is available in the latest release (version 1.2.1).  Let me know if there's any issues.
